impl/maven-cli/src/main/java/org/apache/maven/cling/extensions/ExtensionConfigurationModule.java

```diff
@@ -36,7 +37,9 @@ public class ExtensionConfigurationModule implements Module {
 
     private final CoreExtensionEntry extension;
     private final UnaryOperator<String> callback;
-    private final DefaultInterpolator interpolator = new DefaultInterpolator();
+
+    @Inject
+    private DefaultInterpolator interpolator;
```

Comment:
This is not a component. Just modify the ctor and make it final, and pass it in where instance is created